packages/styles/src/layout/_containers.scss

```diff
@@ -7,26 +7,39 @@
 
 tokens.$default-map: components.$post-container;
 
-.container,
-.container-fluid {
-  width: 100%;
-  margin-inline: auto;
+@each $class in (container, container-fluid) {
+  .#{$class} {
+    width: 100%;
+    margin-inline: auto;
+    --post-container-max-width: #{tokens.get('grid-max-width')};
 
-  @each $breakpoint in map.keys(breakpoints.$grid-breakpoints) {
-    @include media.min($breakpoint) {
-      $gutter: tokens.get('grid-gutter-#{$breakpoint}');
-      $padding: tokens.get('grid-padding-#{$breakpoint}');
+    @each $breakpoint in map.keys(breakpoints.$grid-breakpoints) {
+      @include media.min($breakpoint) {
+        $gutter: tokens.get('grid-gutter-#{$breakpoint}');
+        $padding: tokens.get('grid-padding-#{$breakpoint}');
 
-      --post-grid-gutter-x: #{$gutter};
+        // The `.row` has a negative margin equal to half the gutter size.
+        // To prevent content from overflowing, the container needs to have
+        // a padding of at least half the gutter size to offset this negative margin.
+        padding-inline: max(#{$padding}, calc(0.5 * #{$gutter}));
```

Comment:
This calculation is repeated 3 times, you can create a custom property out of it and reuse it wherever necessary.